app/models/async_transaction/base.rb

```diff
@@ -25,6 +26,14 @@ def parsed_metadata
     validates :transaction_id,
               uniqueness: { scope: :source, message: 'Transaction ID must be unique within a source.' }
 
+    def delete_stale
```

Comment:
As this method can/will `raise` an exception by design, perhaps rename it to `delete_stale!` to imply that.